code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Optimize.java

```diff
@@ -327,4 +327,77 @@ private static void failWithError(final AdobeCallback<?> callback, final AdobeEr
             callbackWithError.fail(error);
         }
     }
+
+    public static void displayPropositions(List<OptimizeProposition> propositions) {
+        trackWithData(
+                generateInteractionXdmForPropositions(
+                        propositions,
+                        OptimizeConstants.JsonValues.EE_EVENT_TYPE_PROPOSITION_DISPLAY));
+    }
+
+    private static Map<String, Object> generateInteractionXdmForPropositions(
+            List<OptimizeProposition> propositions, final String experienceEventType) {
+        final List<Map<String, Object>> decisioningPropositions =
+                getDecisioningPropositions(propositions);
+        final Map<String, Object> experienceDecisioning = new HashMap<>();
+        experienceDecisioning.put(
+                OptimizeConstants.JsonKeys.DECISIONING_PROPOSITIONS, decisioningPropositions);
+        final Map<String, Object> experience = new HashMap<>();
+        experience.put(OptimizeConstants.JsonKeys.EXPERIENCE_DECISIONING, experienceDecisioning);
+
+        final Map<String, Object> xdm = new HashMap<>();
+        xdm.put(OptimizeConstants.JsonKeys.EXPERIENCE, experience);
+        xdm.put(OptimizeConstants.JsonKeys.EXPERIENCE_EVENT_TYPE, experienceEventType);
+        return xdm;
+    }
+
+    private static @NonNull List<Map<String, Object>> getDecisioningPropositions(
+            List<OptimizeProposition> propositions) {
+        final List<Map<String, Object>> decisioningPropositions = new ArrayList<>();
+        for (OptimizeProposition prop : propositions) {
+            decisioningPropositions.add(
+                    new HashMap<String, Object>() {
+                        {
+                            put(
+                                    OptimizeConstants.JsonKeys.DECISIONING_PROPOSITIONS_ID,
+                                    prop.getId());
+                            put(
+                                    OptimizeConstants.JsonKeys.DECISIONING_PROPOSITIONS_SCOPE,
+                                    prop.getScope());
+                            put(
+                                    OptimizeConstants.JsonKeys
+                                            .DECISIONING_PROPOSITIONS_SCOPEDETAILS,
+                                    prop.getScopeDetails());
+                        }
+                    });
+        }
+        return decisioningPropositions;
+    }
+
+    private static void trackWithData(final Map<String, Object> xdm) {
```

Comment:
same comment as above, please move this method to `OptimizeUtils` and change `Offer.java` to use the same. Also please rename this to `track` or `dispatchTrackEvent`